tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -0,0 +1,405 @@
+use crate::{
+    field::RecordFields,
+    fmt::{format, FormatEvent, FormatFields, MakeWriter},
+    layer::{Context, Layer},
+    registry::{LookupMetadata, LookupSpan, SpanRef},
+};
+use std::{cell::RefCell, fmt, io, marker::PhantomData};
+use tracing_core::{
+    span::{Attributes, Id, Record},
+    Event, Subscriber,
+};
+
+/// A [`Layer`] that logs formatted representations of `tracing` events.
+///
+/// [`Layer`]: ../trait.Layer.html
+#[derive(Debug)]
+pub struct FmtLayer<
+    S,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    make_writer: W,
+    fmt_fields: N,
+    fmt_event: E,
+    _inner: PhantomData<S>,
+}
+
+/// A builder for `FmtLayer` that logs formatted representations of `tracing`
+/// events.
+#[derive(Debug)]
+pub struct Builder<
+    S,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    fmt_fields: N,
+    fmt_event: E,
+    make_writer: W,
+    _inner: PhantomData<S>,
+}
+
+impl<S> FmtLayer<S> {
+    /// Creates a [Builder].
+    pub fn builder() -> Builder<S> {
+        Builder::default()
+    }
+}
+
+// This, like the MakeWriter block, needs to be a seperate impl block because we're
+// overriding the `E` type parameter with `E2`.
+impl<S, N, E, W> Builder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Sets a [FormatEvent<S, N>].
+    pub fn with_event_formatter<E2>(self, e: E2) -> Builder<S, N, E2, W>
```

Comment:
The equivalent method on `SubscriberBuilder` is just called
```suggestion
    pub fn fmt_event<E2>(self, e: E2) -> Builder<S, N, E2, W>
```
let's make these more consistent?